src/codegen/compile.ml

```diff
@@ -2735,29 +2738,46 @@ module Object = struct
     )
 
   (* Returns a pointer to the object field (possibly following the indirection) *)
-  let idx_hash env indirect =
+  let idx_hash env m indirect =
     if indirect
-    then Func.share_code2 env "obj_idx_ind" (("x", I32Type), ("hash", I32Type)) [I32Type] (fun env get_x get_hash ->
+    then
+      let name = Printf.sprintf "obj_idx_ind<%d>" m in
+      Func.share_code2 env name (("x", I32Type), ("hash", I32Type)) [I32Type] (fun env get_x get_hash ->
       get_x ^^ get_hash ^^
-      idx_hash_raw env ^^
+      idx_hash_raw env m ^^
       load_ptr ^^ compile_add_const (Int32.mul MutBox.field Heap.word_size)
     )
-    else idx_hash_raw env
+    else idx_hash_raw env m
 
   (* Determines whether the field is mutable (and thus needs an indirection) *)
   let is_mut_field env obj_type s =
     let _, fields = Type.as_obj_sub [s] obj_type in
     Type.is_mut (Type.lookup_val_field s fields)
 
+  (* Computes a lower bound for the linear index of a field in an object *)
```

Comment:
I found "linear index" part a bit confusing. This is computing where to start the linear search for a field, right? I'd say "linear search" instead of "linear index". Maybe "Computes where to start the linear search for a field". 

It would also be good to say here why we can start from an index other than 0, or at least point the reader to the right places for learning about that. (I don't know the answer to this myself)

Reply:
Well, on the surface (Motoko) you index fields by name (which is by hash value in the machine model). Under the hood we have two _parallel arrays_: `[field_hashes_sorted]` and `[field_values]`. See the ASCII art below `module Object`.

Example:
```
| field | hash |
+-----+-----+
|  foo |  789 |
|  bar | 613 |
|  baz | 915 |
```
So when you want to access field `(o : { foo : Int, bar : Int, baz : Int}).foo` with `o` in reality having more fields potentially than the (super)type indicates, you'll find that hash array will be `[613, .., 789, .., 915]` because of sorting. We statically know that the lowest _linear index_ for `foo` is 1, since hash `613` already occupies position 0.

I could say _positional index_ above. Is that okay with you?

Reply:
If you look at all the surrounding  code using `idx*` as "finding a field by hash", you'll probably reconsider that in this context :-)

Reply:
Done, description above extended.